### PR TITLE
Add Enums for nxSocket

### PR DESCRIPTION
**Pull request by danestull**
### What does this Pull Request accomplish?

Adds defined enums from nxSocket for use.

### Why should this Pull Request be merged?

Adding enums helps to complete the XNET use case for socket functions and helps make usage of several functions easier. 

### What testing has been done?

Builds passed, updated test class to use some of the enums for the socket methods and all ran tests passed.

### CODE REVIEW

File: source/codegen/metadata/nixnetsocket/enums.py [hunk 1]
```diff
@@ -1,18 +1,144 @@
...
+            {
+                'name': 'IP_MULTICAST_IF',
+                'value': 5
+            },
+            {
+                'name': 'IP_MULTICAST_TTL',
                 'value': 6
             },
+            {
+                'name': 'IPV6_JOIN_GROUP',
```
Comment (reckenro) [inline source/codegen/metadata/nixnetsocket/enums.py:123]: Thanks for rounding these OptNames out, I think this is all of them from the header and have an email out to see if this is this is all the optnames or not.

I do think we should go ahead and add `IPV6_ADD_MEMBERSHIP` and `IPV6_DROP_MEMBERSHIP` with 12 and 13, respectively. It's duplicated with JOIN/LEAVE_GROUP but I think the codegen will properly add the allow_alias = true to allow the duplicate values in the proto Enum.

File: source/codegen/metadata/nixnetsocket/enums.py [hunk 2]
```diff
@@ -1,18 +1,144 @@
...
+                'name': 'DOWN',
+                'value': 0
+            },
+            {
+                'name': 'UP',
+                'value': 1
+            }
+        ]
+    },
+     'IPStackInffoStringFormat': {
```
Comment (reckenro) [inline source/codegen/metadata/nixnetsocket/enums.py:15]: Typo: 
```suggestion
     'IPStackInfoStringFormat': {
```

File: source/codegen/metadata/nixnetsocket/enums.py [hunk 3]
```diff
@@ -1,18 +1,144 @@
...
+            {
+                'name': 'AF_INET6',
+                'value': 10
+            }
+        ]
+    },
+    'ProtocolFamilies': {
+        'values': [
+            {
+                'name': 'PF_UNSPEC',
```
Comment (reckenro) [inline source/codegen/metadata/nixnetsocket/enums.py:47]: I wonder if it's better for the '..._Unspec' enum values where it's 0 if it's better to just leave it off and have the codegen generate the UNSPECIFIED 0 enum value.

File: source/codegen/metadata/nixnetsocket/enums.py [hunk 4]
```diff
@@ -1,18 +1,144 @@
...
+            {
+                'name': 'TEXT',
+                'value': 1
+            }
+        ]
+    },
+    'AddressFamilies': {
+        'values': [
+            {
+                'name': 'AF_UNSPEC',
```
Comment (reckenro) [inline source/codegen/metadata/nixnetsocket/enums.py:31]: I think we should strip prefixes like 'AF_' from enum value names when their covered by the Enum's name (AddressFamilies covers AF_, ProtocolFamilies covers PF_, etc.)

File: source/codegen/metadata/nixnetsocket/enums.py [hunk 5]
```diff
@@ -1,18 +1,144 @@
...
+                'name': 'JSON',
+                'value': 0
+            },
+            {
+                'name': 'TEXT',
+                'value': 1
+            }
+        ]
+    },
+    'AddressFamilies': {
```
Comment (reckenro) [inline source/codegen/metadata/nixnetsocket/enums.py:28]: I _think_ in general we should shoot for Enum names be singular. So this becomes AddressFamily. Except for Flag enums where more than one might be used in conjunction. @astarche , does this sound right?

File: source/codegen/metadata/nixnetsocket/enums.py (hunk 3 above)
Comment (astarche) [reply source/codegen/metadata/nixnetsocket/enums.py:47]: I like that idea. 

`UNSPECIFIED` is a widely accepted standard/practice. We'll stick with weird driver names pretty far, but when it's this close, I prefer UNSPECIFIED.

File: source/codegen/metadata/nixnetsocket/enums.py (hunk 1 above)
Comment (danestull) [reply source/codegen/metadata/nixnetsocket/enums.py:115]: I added these duplicate values and added the needed suppression to get allow_alias and build working

File: source/codegen/metadata/nixnetsocket/enums.py (hunk 4 above)
Comment (danestull) [reply source/codegen/metadata/nixnetsocket/enums.py]: I've striped these off

File: generated/nixnetsocket/nixnetsocket.proto [hunk 6]
```diff
@@ -42,11 +42,45 @@ enum NiXnetSocketAttribute {
...
+  SOCKET_OPTION_LEVEL_UNSPECIFIED = 0;
+  SOCKET_OPTION_LEVEL_SOCKET = 13;
+}
+
 enum OptName {
+  option allow_alias = true;
   OPT_NAME_UNSPECIFIED = 0;
   OPT_NAME_TCP_NODELAY = 1;
-  OPT_NAME_IP_MULTICAST_TTL  = 6;
+  OPT_NAME_IP_ADD_MEMBERSHIP = 3;
```
Comment (reckenro) [inline generated/nixnetsocket/nixnetsocket.proto:73]: Note: We'll need to handle these new OptName values in the conversion code but I don't think that should be part of this change. Added AB#1889111 to address that.

**Review by reckenro — APPROVED** [review]

**Review by astarche — APPROVED** [review]

**Review by mgumble — COMMENTED** [review]
Looks good!

**Pull request merged**